apiserver/params/network.go

```diff
@@ -69,6 +69,10 @@ type NetworkRoute struct {
 	Metric int `json:"metric"`
 }
 
+// NetworkOrigin specifies where an address comes from, whether it was reported
+// by a provider or by a machine.
+type NetworkOrigin string
```

Comment:
Any reason not to just make this a string on the wire?

Reply:
I wanted to keep type integrity through the stack, so not everything is string string string.